source/qml/Library/RendererQml/AdaptiveCardQmlRenderer.cpp

```diff
@@ -1117,37 +1119,50 @@ namespace RendererQml
+
+				day_Text = "getText(0,2)";
+				month_Text = "getText(3,6)";
+				year_Text = "getText(7,11)";
 				break;
 			}
 			case RendererQml::DateFormat::yymmdd:
 			{
 				StringDateFormat = Formatter() << "yyyy" << dateSeparator << "MMM" << dateSeparator << "dd";
 				inputMask = Formatter() << "xxxx" << dateSeparator << ">x<xx" << dateSeparator << "xx;-";
 				DateRegex = Formatter() << "/^" << year_Regex << dateSeparator << month_Regex << dateSeparator << day_Regex << "$/";
-				//TODO: Change Logic to avoid partial date submit
-				uiDateInput->Property("onTextChanged", Formatter() << "{" << "on_" << uiDateInput->GetId() << "TextChanged(text);" << "if(getText(0,4) === '00' || getText(5,7) === '00' || getText(8,10) === '0000'){selectedDate = '';}else{selectedDate = getText(0,4) + '-' + getText(5,7) + '-' + getText(8,10);}}");
+
+				day_Text = "getText(9,11)";
+				month_Text = "getText(5,8)";
+				year_Text = "getText(0,4)";
 				break;
 			}
 			case RendererQml::DateFormat::yyddmm:
 			{
 				StringDateFormat = Formatter() << "yyyy" << dateSeparator << "dd" << dateSeparator << "MMM";
 				inputMask = Formatter() << "xxxx" << dateSeparator << "xx" << dateSeparator << ">x<xx;-";
 				DateRegex = Formatter() << "/^" << year_Regex << dateSeparator << day_Regex << dateSeparator << month_Regex << "$/";
-				//TODO: Change Logic to avoid partial date submit
-				uiDateInput->Property("onTextChanged", Formatter() << "{" << "on_" << uiDateInput->GetId() << "TextChanged(text);" << "if(getText(0,4) === '00' || getText(5,7) === '00' || getText(8,10) === '0000'){selectedDate = '';}else{selectedDate = getText(0,4) + '-' + getText(8,10) + '-' + getText(5,7);}}");
-                break;
+
+				day_Text = "getText(5,7)";
+				month_Text = "getText(8,11)";
+				year_Text = "getText(0,4)";
+				break;
 			}
 			//Default case: mm-dd-yyyy
 			default:
 			{
-				//TODO: Change Logic to avoid partial date submit
-                uiDateInput->Property("onTextChanged", Formatter() << "{" << "on_" << uiDateInput->GetId() << "TextChanged(text);" << "if(getText(0,2) === '00' || getText(3,5) === '00' || getText(6,10) === '0000'){selectedDate = '';}else{selectedDate = getText(6,10) + '-' + getText(0,2) + '-' + getText(3,5);}}");
 				break;
 			}
 		}
 
+		uiDateInput->AddFunctions(Formatter() << "function setValidDate(dateString)" << "{"
+			<< "var Months = {Jan: 0,Feb: 1,Mar: 2,Apr: 3,May: 4,Jun: 5,July: 6,Aug: 7,Sep: 8,Oct: 9,Nov: 10,Dec: 11};"
+			<< "var d=new Date(" << year_Text << "," << "Months[" << month_Text << "]," << day_Text << ");"
+			<< "if( d.getFullYear().toString() === " << year_Text <<"&& d.getMonth()===Months[" << month_Text << "] && d.getDate().toString()===" << day_Text << ")"
+			<< "{" << "selectedDate = d.toLocaleString(Qt.locale(\"en_US\"),\"yyyy-MM-dd\");" << "}"
+			<< "else { selectedDate = '' };" << "}");
```

Comment:
This function converts the text in the TextField to Date Object. 
To check if the Date Object is valid, we're checking if the day, month and year of the Date object is the same as the input used.